kong/pdk/node.lua

```diff
@@ -49,6 +50,25 @@ end
 
 
 local function new(self)
+  if self and self.configuration and self.configuration.prefix then
```

Comment:
I think it would be a good idea to add some logging for a couple conditions in this change:

* when the filename exists but cannot be read or doesn't contain a valid UUID
* when a new `node_id` is generated, but persisting it to the filesystem fails

Doesn't need to be a super alarming message--probably `ngx.WARN` level? Probably would be helpful to operators in the case of `{{ prefix }}` being r/o or having too restrictive of permissions. WDYT?

Reply:
@flrgh +1, I think we removed the hybrid dp cache file in 3.0; I remember the file open could be a failure with insufficient permissions.
Let's add error logs here and ensure to fall back to random id when failed.

Reply:
@flrgh @mayocream  I've added some logs in some failure scenarios. Let me know if these logs can be improved. 😃

Reply:
@vm-001 Sorry for the late response to this. Assuming this is all temporarily commented-out, it looks pretty good to me.

One suggestion I have is that it would be helpful to include full filesystem paths in each of the unhappy branches so that operators can easily see which file/directory caused something to fail:

```lua
-- NOTE: it's possible that the `read_err` string already contains the full path to `filename`, 
-- making this redundant, so it's best to verify it yourself before making this change
ngx.log(ngx.WARN, "failed to read node id from '", filename, "': ", read_err)
```

Also, please make sure to use varargs in all of your `ngx.log` calls instead of string concatenation:

```lua
-- no
ngx.log(ngx.WARN, "an error happened: " .. err)

-- yes
ngx.log(ngx.WARN, "an error happened: ", err)
```

Reply:
Oh, and this is just a personal opinion of mine, but it can be good to log the side effect when issuing a warning so that operators know exactly what the consequence is (and whether or not they should be concerned about it). To illustrate:

```lua
ngx.log(ngx.WARN, "failed to read node id from '", filename, "': ", err,
                  " (a new node id will be generated)")
```

Some might find this overly verbose, which is a valid take, so I leave it up to your judgment :)

Reply:
@flrgh  Thanks for your suggestion. As too many debug commit here, I made a `git push -f`. Do you have time to review it again?